src/http/api/views/account.posts.view.ts

```diff
@@ -0,0 +1,666 @@
+                this.mapToPostDTO(result, accountId),
+            ),
+            nextCursor: hasMore ? lastResult.published_date : null,
+        };
+    }
+
+    async getPostsByRemoteLookUp(
+        currentContextAccountId: number,
+        currentContextAccountApId: URL,
+        apId: URL,
+        next: URL,
+    ): Promise<Result<AccountPosts, GetPostsError>> {
+        const context = this.fedifyContextFactory.getFedifyContext();
+
+        const documentLoader = await context.getDocumentLoader({
+            handle: 'index',
+        });
+
+        // Lookup actor by handle
+        const actor = await lookupObject(apId, { documentLoader });
+
+        if (!isActor(actor)) {
+            return error('not-an-actor');
+        }
+
+        // Retrieve actor's posts
+        // If a next parameter was provided, use it to retrieve a specific page of
+        // posts. Otherwise, retrieve the first page of posts
+        const result: AccountPosts = {
+            results: [],
+            nextCursor: null,
+        };
+
+        let page: CollectionPage | null = null;
+
+        try {
+            if (next.href !== 'about:blank') {
+                // Ensure the next parameter is for the same host as the actor. We
+                // do this to prevent blindly passing URIs to lookupObject (i.e next
+                // param has been tampered with)
+                // @TODO: Does this provide enough security? Can the host of the
+                // actor be different to the host of the actor's followers collection?
+                const { host: actorHost } = actor?.id || new URL('');
+                const { host: nextHost } = next;
+
+                if (actorHost !== nextHost) {
+                    return error('invalid-next-parameter');
+                }
+
+                page = (await lookupObject(next, {
+                    documentLoader,
+                })) as CollectionPage | null;
+
+                // Check that we have a valid page
+                if (!(page instanceof CollectionPage) || !page?.itemIds) {
+                    page = null;
+                }
+            } else {
+                const outbox = await actor.getOutbox();
+
+                if (outbox) {
+                    page = await outbox.getFirst();
+                }
+            }
+        } catch (err) {
+            return error('error-getting-outbox');
+        }
+
+        if (!page) {
+            return error('no-page-found');
+        }
+
+        // Return result
+        for await (const item of page.getItems()) {
+            try {
+                if (!(item instanceof Activity)) {
+                    continue;
+                }
+
+                const object = await item.getObject();
+                const attributedTo = await object?.getAttribution();
+
+                const activity = (await item.toJsonLd({
+                    format: 'compact',
+                    // TODO: Clean up the any type
```

Comment:
We shouldn't introduce more of these comments, we should be cleaning up as we go! `unknown` is the best type to use when we don't know what we're getting back - it means we have to do a few extra checks, but that's good because we're forced to be type-safe and avoid unexpected runtime errors

Reply:
Cleaned this up.